lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -1699,6 +1810,132 @@ def _validate_valid_in(self):
         return self.cert.get_notBefore(), valid_in_date, self.cert.get_notAfter()
 
 
+class EntrustCertificate(Certificate):
+    """Retrieve a certificate using Entrust (ECS)."""
+
+    def __init__(self, module, backend):
+        super(EntrustCertificate, self).__init__(module, backend)
+        self.trackingId = None
+        self.notAfter = self.get_relative_time_option(module.params['entrust_not_after'], 'entrust_not_after')
+        try:
+            self.ecs_client = ECSClient(
+                entrust_api_user=module.params.get('entrust_api_user'),
+                entrust_api_key=module.params.get('entrust_api_key'),
+                entrust_api_cert=module.params.get('entrust_api_client_cert_path'),
+                entrust_api_cert_key=module.params.get('entrust_api_client_cert_key_path'),
+                entrust_api_specification_path=module.params.get('entrust_api_specification_path')
+            )
+        except SessionConfigurationException as e:
+            module.fail_json(msg='Failed to initialize Entrust Provider: {0}'.format(to_native(e.message)))
+
+    def generate(self, module):
+        if not os.path.exists(self.csr_path):
+            raise CertificateError(
+                'The certificate signing request file {0} does not exist'.format(self.csr_path)
+            )
+
+        if not self.check(module, perms_required=False) or self.force:
+            # Read the CSR that was generated for us
+            body = {}
+            with open(self.csr_path, 'r') as csr_file:
+                body['csr'] = csr_file.read()
+
+            body['certType'] = module.params['entrust_cert_type']
+
+            # Handle expiration (30 days if not specified)
+            expiry = self.notAfter
+            if not expiry:
+                gmt_now = datetime.datetime.fromtimestamp(time.mktime(time.gmtime()))
+                expiry = gmt_now + datetime.timedelta(days=365)
+
+            expiry_iso3339 = expiry.strftime("%Y-%m-%dT%H:%M:%S.00Z")
+            body['certExpiryDate'] = expiry_iso3339
+            body['tracking'] = {
+                'requesterName': module.params['entrust_requester_name'],
+                'requesterEmail': module.params['entrust_requester_email'],
+                'requesterPhone': module.params['entrust_requester_phone']
```

Comment:
```suggestion
                'requesterPhone': module.params['entrust_requester_phone'],
```
It's generally a good idea to add a trailing comma also for the last entry, so when you later add another one in another commit, that change will only be for one new line, and not one line removed and two lines added (because the previously last line got a new comma). And opposed to JSON, Python allows this :)

Reply:
Awesome! I had no idea you could do that in Python. I'm... actually really excited about the impact that has on commit history.